packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -191,3 +186,86 @@ function isSecondsPropertyMisusedInCalendarTriggerInput(
       Object.keys(triggerWithoutChannelId).length > 1)
   );
 }
+
+function validateDateComponents(
+  object: any,
+  components: {
+    day?: boolean;
+    month?: boolean;
+    weekday?: boolean;
+    hour?: boolean;
+    minute?: boolean;
+  }
+) {
+  const expectedProperties = Object.keys(components).filter(key => components[key]);
+  if (
+    expectedProperties.some(
+      component => object[component] === undefined || object[component] === null
+    )
+  ) {
+    const headParams = [...expectedProperties];
+    const tailParam = headParams.pop();
+    const paramsText = headParams.length ? `${headParams.join(', ')} and ${tailParam}` : tailParam;
+    const plural = expectedProperties.length;
+    throw new TypeError(
+      `Parameter${plural ? 's' : ''} ${paramsText} need${
+        plural ? '' : 's'
+      } to have valid values. Found undefined`
+    );
+  }
```

Comment:
This is throwing an error saying all properties are missing even if you're just missing one, let's change it to:

```suggestion
const missingProperties = expectedProperties.filter(
  (component) => object[component] === undefined || object[component] === null
);
if (missingProperties.length) {
  throw new TypeError(
    `Expected values for the following parameter(s), but found null or undefined: ${missingProperties.join(", ")}`
  );
}
```

Reply:
sounds good, I did it in that way because I wanted to keep mostly the same errors that were being thrown before this change, but I agree this is better, I will check it and make this change